### PR TITLE
DNM: Developer docs in progress

### PR DESCRIPTION
**Pull request by BlaineEXE**
Starting to add developer notes to the project. The intent is to bring other devs up to speed more quickly (including myself). Please feel free to make suggestions, add clarifications, or whatever.

### CODE REVIEW

**Comment by BlaineEXE** [comment]
As a specific note, I'd love to get a high-level idea of what `helm-toolkit` is/does. It seems a little like some extra magic on top of the magic that already is Helm.

File: developer/ceph-helm-architecture.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+The architecture of all this stuff
```
Comment (rootfs) [inline developer/ceph-helm-architecture.md]: If this is a placeholder, can you make it in a separate PR once finished?

File: developer/intro-to-helm.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+I'm new to Helm!
```
Comment (rootfs) [inline developer/intro-to-helm.md]: please also add a reference to helm/chart docs for further readings.

**Comment by rootfs** [comment]
thanks for contributing, this is surely a under addressed area we need to polish. 

cc  @dmick @alram
